### PR TITLE
feat: prune sent/cancelled notifications

### LINKED ISSUE #34: Prune send/cancelled notifications
There's currently no mechanism to clean up send/cancelled notifications. Over time the table will grow and grow, so we should probably have some way of cleaning up old entries.

### PR DESCRIPTION
**Pull request by atymic**
Closes #34

### CODE REVIEW

File: src/Console/Commands/PruneScheduledNotifications.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            $query->where('sent_at', '<=', $pruneBeforeDate);
+            $query->orWhere('cancelled_at', '<=', $pruneBeforeDate);
+        });
+
+        $totalDeleted = 0;
+
+        do {
+            $deleted = $notifications->take(1000)->delete();
+            $totalDeleted += $deleted;
+        } while ($deleted !== 0);
```
Comment (litvinjuan) [inline src/Console/Commands/PruneScheduledNotifications.php]: This is not very pretty. Also, why are we taking 1000 at a time for deletion? Shouldn't we simply delete them all at once?
Plus: counter is not necessary in my opinion. Could simply use $notifications->count()

File: src/Console/Commands/PruneScheduledNotifications.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+        if ($pruneDays === null) {
+            $this->error('Pruning of scheduled notifications is disabled');
+
+            return;
+        }
+
+        $pruneBeforeDate = Carbon::now()->subDays($pruneDays);
+
+        $notifications = ScheduledNotification::where(function ($query) {
```
Comment (litvinjuan) [inline src/Console/Commands/PruneScheduledNotifications.php]: You could add a scope to the ScheduledNotification model and just do something like
$notifications = ScheduledNotification::prunnable()

File: src/Console/Commands/PruneScheduledNotifications.php (hunk 1 above)
Comment (atymic) [reply src/Console/Commands/PruneScheduledNotifications.php]: > Also, why are we taking 1000 at a time for deletion? Shouldn't we simply delete them all at once?

From previous experience, there is (or at least was) issues with deleting large amounts of records. I'll try find the docs/bug report 

> counter is not necessary in my opinion. Could simply use $notifications->count()

That would either require us to load all of the scheduled notifications into memory (bad idea) or have to do another table scan to query the count separately (slower)

File: src/Console/Commands/PruneScheduledNotifications.php (hunk 2 above)
Comment (atymic) [reply src/Console/Commands/PruneScheduledNotifications.php]: Thanks for the suggestion, might move the logic to the model.

File: src/Console/Commands/PruneScheduledNotifications.php (hunk 1 above)
Comment (litvinjuan) [reply src/Console/Commands/PruneScheduledNotifications.php]: If you do find that bug report, please do share

Comment (atymic) [reply src/Console/Commands/PruneScheduledNotifications.php]: Here's an example of the issue
https://github.com/laravel/telescope/issues/536

I've had it happen with a table containing ~250k records in the past

**Comment by atymic** [comment]
ping @thomasjohnkane for review when you have a chance :)

**Pull request merged**